### PR TITLE
Implements #442

### PR DESCRIPTION
**Pull request by msoeken**

### CODE REVIEW

File: Standard/src/Canon/Combinators/SinglyControlled.qs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    /// ApplyToEachCA(
+    ///     SinglyControlled(IncrementByInteger)(_, (1, target)),
+    ///     controls)
+    /// );
+    /// ```
+    ///
+    /// # See Also
+    /// - Microsoft.Quantum.Canon.SinglyControlledA
+    function SinglyControlled<'T>(op : 'T => Unit is Ctl) : (Qubit, 'T) => Unit is Ctl {
+        return ApplySinglyControlled(op, _, _);
```
Comment (cgranade) [inline Standard/src/Canon/Combinators/SinglyControlled.qs:30]: You may be able to write this as a lambda now:

```suggestion
        return (ctrl, originalInput) => Controlled op([ctrl], originalInput);
```

Not sure if that generates the right characteristics, as I've not tested that yet, sorry.

File: Standard/tests/Canon/Combinators/SinglyControlled.qs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Tests {
+    open Microsoft.Quantum.Arithmetic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Intrinsic;
+
+    @Test("ToffoliSimulator")
+    operation TestSinglyControlled() : Unit {
```
Comment (cgranade) [inline Standard/tests/Canon/Combinators/SinglyControlled.qs:11]: For testing, this is great, but I may also suggest an AssertOperationsEqualReferenced call for something like `H`:

```qsharp
AssertOperationsEqualReferenced(2,
     (qs) => SinglyControlled(H)(qs[0], qs[1]),
     (qs) => Controlled H([qs[0]], qs[1])
);
```

**Review by cgranade — APPROVED** [review]
Approved with minor comments. Thanks!

File: Standard/src/Canon/Combinators/SinglyControlled.qs (hunk 1 above)
Comment (msoeken) [reply Standard/src/Canon/Combinators/SinglyControlled.qs]: That's awesome! And the recent commit confirms it works.

**Pull request merged**